civis/io/_files.py

```diff
@@ -225,15 +225,18 @@ def file_to_civis(buf, name, api_key=None, client=None, **kwargs):
 
     # we should only pass _file_to_civis a file-like object that is
     # on disk, seekable and at position 0
-    if not isinstance(buf, io.BufferedReader) or buf.tell() != 0:
+    if not isinstance(buf, (io.BufferedReader, io.TextIOWrapper)) or \
+            buf.tell() != 0:
+        # read some bytes to determine proper mode for writing
+        mode = 'w'
+        data = buf.read(5)
```

Comment:
would `buf.read(0)` work?   if so, you could put that in the `isinstance` and get rid of this line and  `fout.write(data)` below.

Reply:
that works. good idea.